src/RawPOSIX/src/safeposix/syscalls/sys_calls.rs

```diff
@@ -168,13 +179,15 @@ impl Cage {
             sigset: newsigset,
             main_threadid: interface::RustAtomicU64::new(0),
             interval_timer: interface::IntervalTimer::new(child_cageid),
+            vmmap: interface::RustLock::new(Vmmap::new()), // Initialize empty virtual memory map for new process
```

Comment:
Why initialize an empty vmmap for child? Can't we just clone parent's vmmap?

Reply:
is the vmmap cloneable and are we sure theyre deep copies? we do need to do this one way or another.